### ISSUE TITLE
Rhizome governance - main sections

### ISSUE THREAD

**Issue by Gab-pm** [okp4/dataverse-portal#105]
# Feature: Rhizome governance - main sections

> **As a** user of the Data Space,
> **I want** to see all the sections of the governance rules,
> **So that** I understand the terms and conditions of the Data Space

### Scenario 1: Default display (desktop)

**Given that** I am a user of the web portal (unauthenticated or authenticated),
**And** I am using a *desktop* device,
**When** I am on the Rhizome Governance page
**Then** I should see a proper description of the different sections that appears as presented in the mockup below

### Scenario 2: Mobile Display

**Given that** I am a user of the web portal (authenticated or unauthenticated),
**When** I am on the Rhizome governance page
**And** I use a mobile device
**Then** I should see mobile friendly display (See mock up)

### Scenario 3: Internationalization

**Given that** I am a user of the web portal (authenticated or unauthenticated),
**When** the selected language is English, French, or German
**Then** title and sections should be displayed in the adequate selected language (See config)

#### Config example

En
> Rhizome | Governance
> Identity Management
> Governance Model
> Datasets Management
> Services Management
> Business Model

Fr
> Rhizome | Gouvernance
> Gestion des Identités
> Modèle de Gouvernance
> Gestion des Jeux de Données
> Gestion des Services
> Modèle Économique

Gr
> Rhizome | Governance
> Verwaltung von Identitäten
> Modell der Governance
> Verwaltung von Datensätzen
> Verwaltung von Dienstleistungen
> Wirtschaftliches Modell

## Requirements

None.

#### Desktop

![image.png](https://images.zenhubusercontent.com/63578bc0de26436874ae0864/4df44f06-b315-4042-8439-80be4978fc21)

#### Mobile

![image.png](https://images.zenhubusercontent.com/63578bc0de26436874ae0864/2ffa66c7-53f4-40fe-985f-5a4072342213)

### Environment

- staging URL: <https://app.staging.okp4.network/>

## Additional informations

None.

**Comment by MdechampG** [comment]
I have written in my notes (in November) `/governance/ID`
As it is not directly to the description of a Data Space, it is not necessary to put it in the dataverse.
What do you think @ccamel ?

**Comment by MdechampG** [comment]
@ccamel @fredericvilcot could we discuss it today ?

**Comment by ccamel** [comment]
Sure @MdechampG.

**Comment by 0xLodz** [comment]
@SpicySalm0n the title is composed of the label of the data space element (`rhizome` on the mockup), separated from the word `governance` by a `|` pipe.

##### Screenshot example for mobile version
![image](https://user-images.githubusercontent.com/8344874/231427122-f0daadc9-c6c9-47f0-996c-605becc506be.png)

What happens when the label contains several words and much more characters? Is it truncated? If yes, what is it's maximum length?
Can the pipe be positioned at the beginning or a end of a line ?

Thanks

**Comment by SpicySalm0n** [comment]
@0xLodz 
the best is to switch on 2 lines, the user can arrive from a web link, it's preferable that the name of the data space is displayed in full.
The pipe can be at the end of the line, no word can be truncated.

<img width="511" alt="Capture d’écran 2023-04-12 à 13 23 41" src="https://user-images.githubusercontent.com/115076387/231442949-9f57f30b-bb91-4b0b-a7c1-16c9dde13a8e.png">

**Comment by MdechampG** [comment]
Decision about the URL of the page : `/dataverse/dataspace/ID/governance`

**Comment by lolottetheclash** [comment]
According to the discussion with @ccamel , it was decided that when a user chooses a `section` as well as a `sub-section`, the url will carry these indications to be shared correctly, like: `/dataverse/dataspace/ID/governance/Section1/SubSection1`
`Section1` & `SubSection1` are both elements ids, like:
```
ex:Section1 a meta:Section, owl:NamedIndividual ;
  core:hasTitle "Datasets Management" ;
  core:hasDescription "Specifies the governance rules regarding the management of datasets." ;
  meta:hasNumber "1.1" ;
  core:isPartOf ex:Chapter1 .
```
Do you agree @MdechampG ?

**Comment by 0xLodz** [comment]
The ID in the URL scheme allow us to find the right dataspace item. Is it possible that no dataspace item are found for this ID? If yes, what should we display to the user?

**Comment by lolottetheclash** [comment]
> The ID in the URL scheme allow us to find the right dataspace item. Is it possible that no dataspace item are found for this ID? If yes, what should we display to the user?

We'll display a 404 page if resource is not found. There is an issue created for this :)

**Comment by 0xLodz** [comment]
@SpicySalm0n  Yes that's totally makes sense. I was wondering that because there is a back arrow button for the mobile Explore page. We might harmonise both page? What do you think? Thanks

**Comment by SpicySalm0n** [comment]
> @SpicySalm0n Yes that's totally makes sense. I was wondering that because there is a back arrow button for the mobile Explore page. We might harmonise both page? What do you think? Thanks

for the catalog detail pages it's a little more complex since the user can arrive from an external link to the portal but the back arrow button brings the user to the Dataverse (Explore). On the other hand, if he's coming from his external link, and he uses the native back button of his app, he will return to his web search or the place from which he clicked on the external link...
That's why the explorer back arrow button have the "dataverse" wording added
I don't think we should remove it for this particular case

**Comment by 0xLodz** [comment]
@SpicySalm0n @MdechampG what to do when section or subsection texts are very large?
Thanks.

<img width="328" alt="image" src="https://user-images.githubusercontent.com/8344874/232993599-59a9b44a-41f3-4498-b1c4-e37ef127868c.png">

**Comment by ChaimaaZaoui** [comment]
@0xLodz I edited the english text in internationalisation

**Comment by 0xLodz** [comment]
@ChaimaaZaoui on the screenshot I changed the section title for testing.
I added all mocked data you provided in [this file](https://github.com/okp4/dataverse-portal/blob/d9af53c038e08f4a9c7f081495e8dc37be5639b8/src/ui/page/dataverse/dataspace/governance/mockedData.ts).

Could we have new sections and subsections in the future that are larger than the one already provided? Thanks

**Comment by ChaimaaZaoui** [comment]
@0xLodz for now the length of text is not limited, but this is something we are going to work on when defining "create data space" module

**Comment by MdechampG** [comment]
You're right @ChaimaaZaoui but even if we restrain the number of characters in the UI "Create a Data Space", it will be possible (through other interfaces than the portal) to have long titles. 
I suggest to have an ellipse. @SpicySalm0n what should be the maximum size that we display ?

**Comment by SpicySalm0n** [comment]
For the Section and Subsection titles we could put an ellipse with one line or two lines, both are ok for me:

Sections:

One line:
<img width="271" alt="Capture d’écran 2023-04-19 à 11 58 34" src="https://user-images.githubusercontent.com/115076387/233039997-6484a97d-d9ea-4589-a62e-08b5981fbcae.png">

Two lines: 
<img width="328" alt="Capture d’écran 2023-04-19 à 11 57 30" src="https://user-images.githubusercontent.com/115076387/233039697-8965acf1-4eaa-4e7b-b7a2-392dbc8bd0f6.png">

Subsections:

One line:
<img width="462" alt="Capture d’écran 2023-04-19 à 11 46 50" src="https://user-images.githubusercontent.com/115076387/233038774-9e322574-4464-4a4a-b1c9-b307aad2f6df.png">

Two lines:
<img width="178" alt="Capture d’écran 2023-04-19 à 11 54 23" src="https://user-images.githubusercontent.com/115076387/233038837-e5258efb-095f-474e-8766-be5221cd6528.png">

**Comment by SpicySalm0n** [comment]
in any case the text must not exceed the width of the 3 columns of the layout

**Comment by 0xLodz** [comment]
I implemented the one line solution. Don't hesitate to ask for changes if necessary. Thanks

**Comment by ChaimaaZaoui** [comment]
Test executed in firefox
- S1 : Pass
- S2 : not run, to be tested in #200 
- S3 : Pass 
- S4 : Pass
- S5 : Pass
- S6 : not run, to be tested in #207 
- S7 : Pass
- S8 : not run, to be tested in #207
